### PR TITLE
Migrate pop over to compose ANDROID-11502

### PR DESCRIPTION
**Pull request by fjquileze**
🎟️ Jira tickets

[Task](https://jira.tid.es/browse/ANDROID-11502)

### :goal_net: What's the goal?
Migrate XML PopOver implementation to Compose

### :construction: How do we do it?
Added new PopOver composable

### ☑️ Checks
- [x] I updated the documentation, including readmes and wikis. If this is a breaking change, update [UPGRADING.md](../UPGRADING.md) to inform users how to proceed. If no updates are necessary, indicate so.
- [ ] Tested with dark mode.
- [x] Tested with API 23.

### :test_tube: How can I test this?
_If it cannot be tested explain why._
- [ ] 🖼️ Screenshots/Videos
- [ ] Mistica App QR or download link
- [ ] Reviewed by Mistica design team
- [ ] ...

### CODE REVIEW

**Comment by DevPabloGarcia** [comment]
Testing the app I found some problems. 
* When the title text is too long, it shows under the "X" close button and description is not shown.
* When the title fit the popover but the description is too long, the pop over doesn't expand.

Checking the xml behavior, title and description could be fit in multiple lines.
Screenshot_20230522_124612
Screenshot_20230522_124840

**Review by DevPabloGarcia — CHANGES_REQUESTED** [review]
I found some errors, please check it 😄

**Comment by gmerinojimenez** [comment]
Hi, minor, but the arrow is not starting from the component, looks like there is an extra space, no?

telegram-cloud-photo-size-4-6005900280828377764-x

File: library/src/main/java/com/telefonica/mistica/compose/popover/PopOver.kt [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                    }
+                    Column(
+                        modifier = Modifier
+                            .wrapContentHeight()
+                            .padding(start = 16.dp)
+                    ) {
+                        Text(
+                            text = title,
+                            style = MisticaTheme.typography.preset3
+                        )
```
Comment (gmerinojimenez) [inline library/src/main/java/com/telefonica/mistica/compose/popover/PopOver.kt:101]: I think this should be limited to only one line

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/PopOvers.kt [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import com.telefonica.mistica.compose.theme.MisticaTheme
+
+@Composable
+fun PopOvers() {
+
+    var title: String by remember { mutableStateOf("Title of the pop over") }
+    var subtitle: String by remember { mutableStateOf("Popover long description text") }
+    var addImage by remember { mutableStateOf(true) }
+    val orientationItems = remember { listOf("AUTO", "TOP", "BOTTOM") }
+    var selectedItemIndex by remember { mutableStateOf<Int?>(0) }
```
Comment (gmerinojimenez) [inline catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/PopOvers.kt:37]: maybe we can keep an enum to improve the readability :)
```kotlin
enum PopoverPosition {
    Auto, Top, Bottom
}
```

**Review by gmerinojimenez — CHANGES_REQUESTED** [review]
Nice job! please check the small required changes and you have it :)

**Comment by fjquileze** [comment]
> 
fixed! 👍

File: library/src/main/java/com/telefonica/mistica/compose/popover/PopOver.kt [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+                Image(
+                    painter = painterResource(id = R.drawable.icn_cross),
+                    contentDescription = null,
+                    modifier = Modifier
+                        .wrapContentSize()
+                        .align(Alignment.TopEnd)
+                        .clickable {
+                            window?.dismiss()
+                        }
+                )
```
[the remark this answers is not in the document]
Comment (fjquileze) [reply library/src/main/java/com/telefonica/mistica/compose/popover/PopOver.kt:119]: fixed! 👍

File: library/src/main/java/com/telefonica/mistica/compose/popover/PopOver.kt (hunk 1 above)
Comment (fjquileze) [reply library/src/main/java/com/telefonica/mistica/compose/popover/PopOver.kt:101]: I checked the figma documentation about the pop overs and has to be limited to 2, I'm gonna change it

File: catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/PopOvers.kt (hunk 2 above)
Comment (fjquileze) [reply catalog/src/main/java/com/telefonica/mistica/catalog/ui/compose/components/PopOvers.kt:37]: Good point, I'm going to change it

**Review by gmerinojimenez — APPROVED** [review]
Good job!

**Review by DevPabloGarcia — APPROVED** [review]
Awesome!

**Review by aweell — APPROVED** [review]

**Pull request merged**